File: tennis-wave-frontend/src/types/tennisMatch.ts
// src/types/tennisMatch.ts
export interface TennisMatch {
    id: number;
    title: string;
    description: string;
    location: string;
    matchTime: string;
    latitude?: number;
    longitude?: number;
    matchType: string;
    skillLevel: string;
    maxParticipants: number;
    currentParticipants: number;
    status: string;
    createdAt: string;
    creatorId: number;
    creatorName: string;
    creatorAvatar: string;
}

export interface MatchParticipant {
    participantId: number;
    matchId: number;
    userId: number;
    userName: string;
    joinDate: string;
    status: 'Pending' | 'Confirmed' | 'Rejected';
}

export interface CreateMatchDto {
    title: string;
    description: string;
    location: string;
    matchTime: string;
    latitude?: number;
    longitude?: number;
    matchType: string;
    skillLevel: string;
    maxParticipants: number;
}

export interface JoinMatchDto {
    matchId: number;
}

export interface MatchParticipant {
    participantId: number;
    matchId: number;
    userId: number;
    userName: string;
    joinDate: string;
    status: 'Pending' | 'Confirmed' | 'Rejected';
}

File: tennis-wave-api/Models/Entities/TennisMatch.cs
using System;
using System.Collections.Generic;

namespace tennis_wave_api.Models.Entities;

public partial class TennisMatch
{
    public int Id { get; set; }

    public string Title { get; set; } = null!;

    public string Description { get; set; } = null!;

    public DateTime MatchTime { get; set; }

    public string Location { get; set; } = null!;

    public double? Latitude { get; set; }

    public double? Longitude { get; set; }

    public string MatchType { get; set; } = null!;

    public string SkillLevel { get; set; } = null!;

    public int MaxParticipants { get; set; }

    public int CurrentParticipants { get; set; }

    public int CreatorId { get; set; }

    public string Status { get; set; } = null!;

    public DateTime CreatedAt { get; set; }

    public DateTime? UpdatedAt { get; set; }
    
    public virtual User Creator { get; set; } = null!;
    public virtual ICollection<MatchParticipant> Participants { get; set; } = new List<MatchParticipant>();
}
